File: packages/provide-uterm-go/vt/charsets.go
//
// SPDX-FileCopyrightText: Copyright (c) 2025-2026 provide.io llc. All rights reserved.
// SPDX-License-Identifier: AGPL-3.0-or-later
//

package vt

// charsetMaps maps charset designator codes to their translation tables,
// mirroring pyte.charsets.MAPS. The tables themselves live in the
// generated charset_tables.go.
var charsetMaps = map[string]*charsetMap{
	"B": &lat1Map,  // Latin-1.
	"0": &vt100Map, // VT100 special graphics.
	"U": &ibmpcMap, // IBM codepage 437.
	"V": &vax42Map, // VAX42.
}

File: packages/provide-uterm-csharp/src/Provide.Uterm/Vt/Charsets.cs
//
// SPDX-FileCopyrightText: Copyright (c) 2025-2026 provide.io llc. All rights reserved.
// SPDX-License-Identifier: AGPL-3.0-or-later
//

namespace Provide.Uterm.Vt;

internal static class Charsets
{
    internal static readonly Dictionary<string, int[]> Maps = new()
    {
        ["B"] = CharsetTables.Lat1Map,
        ["0"] = CharsetTables.Vt100Map,
        ["U"] = CharsetTables.IbmpcMap,
        ["V"] = CharsetTables.Vax42Map,
    };
}
